### PR TITLE
Fix script compile errors not being flushed to log file

### PR DESCRIPTION
**Pull request by wolf109909**
i can't spell before :(

### CODE REVIEW

File: NorthstarDLL/squirrel.cpp [hunk 1]
```diff
@@ -359,7 +359,11 @@ void __fastcall ScriptCompileErrorHook(HSquirrelVM* sqvm, const char* error, con
 	{
 		// kill dedicated server if we hit this
 		if (IsDedicatedServer())
+		{
+			// flush the logger before we abort so debug things gets saved to log file
```
Comment (GeckoEidechse) [inline NorthstarDLL/squirrel.cpp:363]:
```suggestion
			// flush the logger before we abort so debug things get saved to log file
```

**Comment by wolf109909** [comment]
uwu

**Review by emma-miler — APPROVED** [review]

**Comment by GeckoEidechse** [comment]
@pg9182 wanna test this with Docker?

**Review by emma-miler — APPROVED** [review]
Code looks good. Untested, but would work in theory.

**Review by GeckoEidechse — APPROVED** [review]
Confirmed working in testing by inserting syntax error into some Squirrel scripts on purpose

**Before:**
```
...
[21:36:40] [FILESYSTM] [info] Changing mod search path from C:\Program Files (x86)\Steam\steamapps\common\Titanfall2\R2Northstar\mods\Northstar.CustomServers\mod to R2Northstar/mods\Northstar.Custom
[21:36:40] [FILESYSTM] [info] Changing mod search path from C:\Program Files (x86)\Steam\steamapps\common\Titanfall2\R2Northstar\mods\Northstar.Custom\mod to R2Northstar/mods\Northstar.CustomServers
[21:36:40] [FILESYSTM] [info] Changing mod search path from C:\Program Files (x86)\Steam\steamap
```

**After:**
```
...
[22:47:38] [FILESYSTM] [info] Changing mod search path from C:\Program Files (x86)\Steam\steamapps\common\Titanfall2\R2Northstar\mods\Northstar.Custom\mod to R2Northstar/mods\Northstar.CustomServers
[22:47:38] [SCRIPT SV] [error] COMPILE ERROR expected ")", found identifier "asdasd"
[22:47:38] [SCRIPT SV] [error] lobby/_lobby.gnut line [22] column [33]
```

**Pull request merged**